types/react-router/index.d.ts

```diff
@@ -61,7 +61,7 @@ declare module 'react-router' {
 
   interface RouteProps {
     location?: H.Location;
-    component?: React.SFC<RouteComponentProps<any> | void> | React.ComponentClass<RouteComponentProps<any> | void>;
+    component?: React.SFC<RouteComponentProps<any> | void | {}> | React.ComponentClass<RouteComponentProps<any> | void | {}>;
```

Comment:
I said that `RouteComponentProps` subtypes `{}`, which is the opposite of what you're trying there.
```ts
declare var props: RouteComponentProps<any>;
const foo: {} = props;
```